README.md

```diff
@@ -1,16 +1,1831 @@
+  "input": {
+    "func": "add_one",
+    "args": [
+      42
+    ]
+  }
+}
+```
+
+# 4 Authorization
+
+An [Authorization] is cryptographically signed data set. It represents an authorization to run [Task]s in that are included in `scope` data set.
+
+## 4.1 Schema
+
+```ipldsch
+type Authorization struct {
+  # Authorization is denoted by the set of links been authorized
+  scope   [&Any] (implicit [])
+  # Scope signed by the invoker
+  s       VarSig
+}
+```
+
+### 4.2 Fields
+
+#### 4.2.1 Authorization Scope
+
+The `scope` field MUST be a set of links been authorized. It SHOULD be encoded as an alphabetically ordered list without duplicates.
+
+If `scope` field is omitted, it is implicitly a an empty list and has no practical use as it authorizes nothing.
+
+### 4.2.2 Signature
+
+The `s` field MUST contain a [Varsig] of the [CBOR] encoded `scope` field.
+
+## 4.3 DAG-JSON Example
+
+```json
+{
+  "scope": [
+    {
+      "/": "bafyreigfusg7tgegda7pxdn7px5vs7wxqk5nybr5ewjht4xas357x6ryia"
+    },
+    {
+      "/": "bafyreid32aunjg6g6buib7bva2sa5ni4c6eaftgykyh6z6rnhcjxh4yk2y"
+    }
+  ],
+  "s": {
+    "/": {
+      "bytes": "7aEDQJBsO/9rkcgQQ9qvSC0E60MdqGjpIHXuc4VbkfjEeYR7iAvAY+0QEN9kAFcwh6/kww4bK/I7xsqLzO5kG0x+2Qs"
+    }
+  }
+}
+```
+
+# 5 Invocation
+
+As [noted in the introduction][lazy-vs-eager], there is a difference between a reference to a function and calling that function. The [Invocation] is an instruction to the [Executor] to perform enclosed [Task]. [Invocation]s are not executable until they have been provided provable authority (in form of UCANs in the `prf` field) and an [Authorization] (in the `auth` field) from the [Invoker].
+
+If `auth` field MUST be set to the [Authorization] (link) which contains `run` in it's scope field. The Invocation of the [Task] with an [Authorization] which does not includes [Task] in the `scope` MUST be considered invalid.
+
+## 5.1 Schema
+
+```ipldsch
+type Invocation struct {
+  v       SemVer
+
+  run     &Task
+  # Receipt of the invocation that caused this invocation
+  cause   optional &Invocation
+  # Task authorization.
+  auth    &Authorization
+
+  meta    {String : any}
+
+  prf     [&UCAN]
+}
+type SemVer string
```

Comment:
Very much a tangent, but do we want to define a structured SemVer type?

```haskell
type SemVer struct {
  major Integer
  minor Integer
  patch Integer
  tag   optional String
} representation tuple
```

Reply:
I don't think tuples representation allows optional fields, that said it's fine by me. There is also [stringjoin representation](https://ipld.io/docs/schemas/features/representation-strategies/#struct-stringjoin-representation) worse considering.

Reply:
Does `stringjoin` respect optionals? If so:

```haskell
type SemVer struct {
  num NumVer
  tag optional String
} respresentation stringjoin {
  join "+"
}

type NumVer struct {
  major Integer
  minor Integer
  patch Integer
} respresentation stringjoin {
  join "."
}
```

But leaving it as a string is also fine 🤷‍♀️

Reply:
Yeah I was wondering the same. Personally I have no strong opinion whichever way we is fine by me. I do would like to be consistent between this and UCANs however.

Reply:
Yup agreed. I _think_ the above will render as a string and parse, so it would only be a question for ucan-ipld, right? I'm happy leaving it as a string for now and adding this later though.

Reply:
(Esp since I think it would be backwards compatible)